packages/go/index.ts

```diff
@@ -501,17 +520,54 @@ export async function build({
     throw error;
   } finally {
     try {
-      await cleanupFileSystem(undoFileActions, undoDirectoryCreation);
+      await cleanupFileSystem(
+        undoFileActions,
+        undoDirectoryCreation,
+        undoFunctionRenames
+      );
     } catch (error) {
       console.log(`Build succeeded, but cleanup failed: ${error.message}`);
       debug('Cleanup Error: ' + error);
     }
   }
 }
 
+async function renameHandlerFunction(fsPath: string, from: string, to: string) {
+  let fileContents = (await readFile(fsPath)).toString('utf-8');
+
+  const fromRegex = new RegExp(`\\b${from}\\b`, 'g');
+  fileContents = fileContents.replace(fromRegex, to);
+
+  await writeFile(fsPath, fileContents);
+}
+
+async function getNewHandlerFunctionName(
+  originalFunctionName: string,
+  entrypoint: string
+) {
+  debug(`Found exported function "${originalFunctionName}" in "${entrypoint}"`);
+
+  const pathSlug = entrypoint
+    .replace(/\\/g, '_')
+    .replace(/\//g, '_')
+    .replace(/\[/g, '_')
+    .replace(/\]/g, '_')
+    .replace(/ /g, '_')
+    .replace(/-/g, '_')
+    .replace(/\./g, '_');
```

Comment:
```suggestion
  const pathSlug = entrypoint.replace(/( \\|\/|\]|\[|-|\.)/g, '_')
```

Reply:
Good catch, though you forgot a `|` after the space and before the `\\`.

This could be simplified further by doing:

```javascript
const pathSlug = entrypoint.replace(/[\\/[\] -.]+/g, '_');
```

Also, you probably will want to add a `+` near the end so it replaces multiple consecutive characters with a single `_`:

```javascript
'a\\b/c[d]e f-g.h/-./\\i'.replace(/[\\/[\] -.]+/g, '_') // 'a_b_c_d_e_f_g_h_i'
'a\\b/c[d]e f-g.h/-./\\i'.replace(/[\\/[\] -.]/g, '_') // 'a_b_c_d_e_f_g_h_____i'
```

Reply:
I think we should optimize for legibility on this one. However, I don't think one is necessarily easier to read than the other. With syntax highlighting at least, using `|` a bunch is helpful.

I went with `/(\s|\\|\/|\]|\[|-|\.)/g`. I added a bunch of tests for it.

I don't think it's critical to replace multiple instances. No one is supposed to even see this value.

https://github.com/vercel/vercel/pull/8299/commits/2b4d729f88a044994d026ec0b82bde56a3b6792f